test/results/test_protein_folding_result.py

```diff
@@ -10,66 +10,130 @@
+    test_result_2 = create_protein_folding_result("APRLR", ["", "", "F", "Y", ""], "0011011")
+    test_result_3 = create_protein_folding_result("APRLR", ["", "F", "", "Y", ""], "10110110")
+
+    def test_best_sequence(self):
+        """Tests if the best sequence obtained is the correct one and if
+        it gets passed to the constructor correctly."""
+        # Tests for case 1
+        self.assertEqual(self.test_result_1.best_sequence, "101100011")
+        # Tests for case 2
+        self.assertEqual(self.test_result_2.best_sequence, "0011011")
+        # Tests for case 3
+        self.assertEqual(self.test_result_3.best_sequence, "10110110")
+
+    def test_binary_vector(self):
+        """Tests if the result binary vector is expanded correctly"""
+        # Test for case 1
+        self.assertEqual(
+            self.test_result_1.get_result_binary_vector(),
+            "1******0*************************************"
+            + "*********************************************"
+            + "*******************************************110001*1****",
+        )
+        # Test for case 2
+        self.assertEqual(self.test_result_2.get_result_binary_vector(), "0011****01*1****")
+        # Test for case 3
+        self.assertEqual(self.test_result_3.get_result_binary_vector(), "10**11**0110****")
+
+    def test_side_chain_hot_vector(self):
+        """Tests if the hot vector from the side chain is correct"""
+        # Test for case 1
+        self.assertEqual(
+            self.test_result_1._side_chain_hot_vector,
+            [False, False, False, False, False, False, False],
+        )
+        # Test for case 2
+        self.assertEqual(
+            self.test_result_2._side_chain_hot_vector, [False, False, True, True, False]
+        )
+        # Test for case 3
+        self.assertEqual(
+            self.test_result_3._side_chain_hot_vector, [False, True, False, True, False]
+        )
+
+    def test_unused_qubits(self):
+        """Tests the list of unused qubits"""
+        # Test for case 1
+        max_index_1, used_qubits_1 = 167, [4, 6, 7, 8, 9, 10, 11, 137, 144]
+        expected_unused_qubits_1 = [n for n in range(max_index_1 + 1) if n not in used_qubits_1]
+        self.assertEqual(
+            self.test_result_1._unused_qubits,
+            expected_unused_qubits_1,
+        )
+
+        # Test for case 2
+        max_index_2, used_qubits_2 = 79, [4, 6, 7, 12, 13, 14, 15]
+        expected_unused_qubits_2 = [n for n in range(max_index_2 + 1) if n not in used_qubits_2]
+        self.assertEqual(
+            self.test_result_2._unused_qubits,
+            expected_unused_qubits_2,
+        )
+
+        # Test for case 3
+        max_index_3, used_qubits_3 = 79, [4, 5, 6, 7, 10, 11, 14, 15]
+        expected_unused_qubits_3 = [n for n in range(max_index_3 + 1) if n not in used_qubits_3]
+
+        self.assertEqual(
+            self.test_result_3._unused_qubits,
+            expected_unused_qubits_3,
+        )
```

Comment:
I would suggest not testing all cases in one method. A better pattern would be to have a parametrized test method and provide every test case as data with the `ddt` framework. This comment applies to other tests, too.